packages/@aws-cdk/aws-events/lib/connection.ts

```diff
@@ -0,0 +1,423 @@
+import * as crypto from 'crypto';
+import { IResource, Lazy, Names, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnConnection } from './events.generated';
+
+/**
+ * Supported Authorization Types.
+ */
+export enum AuthorizationType {
+  /** API_KEY */
+  API_KEY = 'API_KEY',
+  /** BASIC */
+  BASIC = 'BASIC',
+  /** OAUTH_CLIENT_CREDENTIALS */
+  OAUTH_CLIENT_CREDENTIALS = 'OAUTH_CLIENT_CREDENTIALS',
+}
+
+/**
+ * Contains the API key authorization parameters to use for the connection.
+ */
+export interface ApiKeyAuthParameters {
+  /**
+   * The name of the API key to use for authorization.
+   */
+  readonly apiKeyName: string;
+
+  /**
+   * The value for the API key to use for authorization.
+   */
+  readonly apiKeyValue: string;
```

Comment:
Should this be a `cdk.SecretValue` instead of a string ?

Have a look at the [GitHub Action for CodePipeline](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-codepipeline-actions/lib/github/source-action.ts) for an example.

Same applies for other secrets below. Wondering if we should set the `apiKeyName`/`username`/`clientId` as secretValues as well ?

Reply:
Let me rephrase:

Password and keys DEFINITELY need to be typed as `cdk.SecretValue`.

Reply:
EventBridge consideres api key names, usernames, and client Ids as non-secret, and their value counterparts secrets

On the other hand, headers, query strings, and body parameters can be classified as secret/non-secret by the customer.